google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageDispatcher.java

```diff
@@ -57,6 +61,9 @@
   @VisibleForTesting static final Duration PENDING_ACKS_SEND_DELAY = Duration.millis(100);
   private static final int MAX_ACK_DEADLINE_EXTENSION_SECS = 10 * 60; // 10m
 
+  private static final ScheduledExecutorService alarmsExecutor =
+      Executors.newScheduledThreadPool(2);
```

Comment:
As per my load tests, this scales well with hundreds of subscriber objects sharing it.